Sources/BaggageContext/Context.swift

```diff
@@ -0,0 +1,287 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Baggage Context open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Swift Baggage Context project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Baggage
+import Logging
+
+/// A default `Context` type.
+///
+/// It is a carrier of contextual `Baggage` and related `Logger`, allowing to log and trace throughout a system.
+///
+/// Any values set on the `baggage` will be made accessible to the logger as call-site metadata, allowing it to log those.
+///
+/// ### Logged Metadata and Baggage Items
+///
+/// Please refer to your configured log handler documentation about how to configure which metadata values should be logged
+/// and which not, as each log handler may handle and configure those differently. The default implementations log *all*
+/// metadata/baggage values present, which often is the right thing, however in larger systems one may want to choose a
+/// log handler which allows for configuring these details.
+///
+/// ### Accepting context types in APIs
+///
+/// It is preferred to accept values of `ContextProtocol` in library APIs, as this yields a more flexible API shape,
+/// to which other libraries/frameworks may pass their specific context objects.
+///
+/// - SeeAlso: `Baggage` from the Baggage module.
+/// - SeeAlso: `Logger` from the SwiftLog package.
+public struct Context: ContextProtocol {
+    /// The `Baggage` carried with this context.
+    /// It's values will automatically be made available to the `logger` as metadata when logging.
+    ///
+    /// Baggage values are different from plain logging metadata in that they are intended to be
+    /// carried across process and node boundaries (serialized and deserialized) and are made
+    /// available to instruments using `swift-distributed-tracing`.
+    public var baggage: Baggage
+
+    public var logger: Logger {
+        get {
+            return self._logger.with(self.baggage)
+        }
+        set {
+            self._logger = newValue
+        }
+    }
```

Comment:
The `Context`, previously proposed a few times by various people as "DefaultContext".

It automatically handles the "right logger" creation, so most users if they create contexts will not have to deal with this complexity.

Framework authors which do not use `Context` but have `MyLib.Context: ContextProtocol` have to implement it correctly and we're okey with that -- that's far less people than "every user" :)

Reply:
setting the logger beyond the constructor intentional?

Reply:
Yes, to allow logLevel changes.

Reply:
Thanks for chiming in Adam, the package naming is also important indeed. I would like to avoid "plural names" though. There technically exists precedence for `Swift...` prefixed modules because `SwiftProtobuf` does so, but yes you're right it's not a pattern we follow.

I very much like @slashmo  proposal where the package name is more "functional" rather than just repeating the same words again, we brainstormed this a bit more and I would like to propose the following which I think addresses all concerns:

- `.package("https://github.com/apple/swift-context-propagation.git", from: "1.0.0")`
  - module name: `import ContextPropagation`
  - includes `protocol BaggageContext { Baggage Logger }` and `DefaultContext`, and friends (the log handler etc)
  - depends on `SwiftLog`
- `.package("https://github.com/apple/swift-context-propagation-baggage.git", from: "1.0.0")`
    - libraries which do not want to depend on logging can use this if they still need to carry a baggage somehow (e.g. NIO this way is able to not have to depend on logging, but offer `context.baggage` itself)
  - module name: `import ContextPropagationBaggage`
  - includes `struct Baggage` and the related key types etc
  - zero dependencies


Please note that repository names are subject to input from Apple teams and management etc, so I can't commit to those, but those would seem a very good balanced tradeoff. Not squatting the general term "context" and not looking weird "swift-baggage" which is a bit "what does this repo even do?" for the uninitiated. It also solves our module name != main type requirement mentioned up-thread here.

Reply:
I believe this was added to the wrong comment ;)

Are we still sure there needs to be two repos instead of just one?

Reply:
Whoops yes, let me add it to the right one... too many tabs :~